spec/lib/task-spec.js

```diff
@@ -242,6 +242,39 @@ describe("Task", function () {
             expect(task.options.nested3).to.equal(definition.options.sourceValue);
         });
 
+        it("should render iteration templates", function() {
+            definition.options = {
+                testList: [
+                    {
+                        name: 'item 1'
+                    },
+                    {
+                        name: 'item 2'
+                    },
+                    {
+                        name: 'item 3'
+                    }
+                ],
+                testVal: '{{#options.testList}}{{ name }}.{{/options.testList}}'
+            };
+            var task = Task.create(definition, {}, {});
+            var tempList = _.transform(definition.options.testList, function (result, n) {
+                result.push(n.name);
+            });
+            expect(task.options.testVal).to.equal(tempList.join('.') + '.');
+        });
+
+        it("should render condition templates", function() {
+            definition.options = {
+                testSrc1: 'Test source 1 exist',
+                testVal1: '{{#options.testSrc1}}{{ options.testSrc1 }}{{/options.testSrc1}}',
+                testVal2: '{{#options.testSrc2}}{{ options.testSrc2 }}{{/options.testSrc2}}'
+            };
+            var task = Task.create(definition, {}, {});
+            expect(task.options.testVal1).to.equal(definition.options.testSrc1);
+            expect(task.options.testVal2).to.equal('');
```

Comment:
Hm, okay, so general question to you here: Currently in the task rendering code we have decided to fail task validation if a render string references a value that does not exist, whereas with normal hogan rendering, it looks like it will just bypass that. Do you think we should care about enforcing that behavior for complex templates, or just decide to render certain values as empty silently as this is doing?